src/plugins/vis_type_timeseries/server/lib/search_strategies/strategies/abstract_search_strategy.ts

```diff
@@ -33,41 +34,50 @@ import { IndexPatternsFetcher } from '../../../../../data/server';
  * This will be replaced by standard KibanaRequest and RequestContext objects in a later version.
  */
 export type ReqFacade = FakeRequest & {
+  requestContext: RequestHandlerContext;
   framework: Framework;
   payload: unknown;
   pre: {
     indexPatternsService?: IndexPatternsFetcher;
   };
   getUiSettingsService: () => IUiSettingsClient;
   getSavedObjectsClient: () => SavedObjectsClientContract;
-  server: {
-    plugins: {
-      elasticsearch: {
-        getCluster: () => {
-          callWithRequest: (req: ReqFacade, endpoint: string, params: any) => Promise<any>;
-        };
-      };
-    };
-  };
   getEsShardTimeout: () => Promise<number>;
 };
 
 export class AbstractSearchStrategy {
-  public getCallWithRequestInstance: (req: ReqFacade) => LegacyAPICaller;
-  public getSearchRequest: (req: ReqFacade) => any;
-
-  constructor(
-    server: any,
-    callWithRequestFactory: (server: any, req: ReqFacade) => LegacyAPICaller,
-    SearchRequest: any
-  ) {
-    this.getCallWithRequestInstance = (req) => callWithRequestFactory(server, req);
+  public searchStrategyName!: string;
+  public indexType?: string;
+  public additionalParams: any;
 
-    this.getSearchRequest = (req) => {
-      const callWithRequest = this.getCallWithRequestInstance(req);
+  constructor(name: string, type?: string, additionalParams: any = {}) {
+    this.searchStrategyName = name;
+    this.indexType = type;
+    this.additionalParams = additionalParams;
+  }
 
-      return new SearchRequest(req, callWithRequest);
-    };
+  async search(core: CoreSetup, req: ReqFacade, bodies: any[], options = {}) {
+    const deps = (await core.getStartServices())[1] as VisTypeTimeseriesPluginStartDependencies;
```

Comment:
nit: I think passing the `core` explicitly complicates, it's already presented in `req.framework.core`

```suggestion
async search(req: ReqFacade, bodies: any[], options = {}) {
    const [, deps] = await req.framework.core.getStartServices();
```

This will make unnecessary imports of `CoreSetup` & `VisTypeTimeseriesPluginStartDependencies`

Reply:
done